atom/browser/mac/atom_application.mm

```diff
@@ -29,10 +29,16 @@ - (void)setHandlingSendEvent:(BOOL)handlingSendEvent {
 }
 
 - (void)setCurrentActivity:(NSString*)type
-              withUserInfo:(NSDictionary*)userInfo {
+              withUserInfo:(NSDictionary*)userInfo
+            withWebpageURL:(NSURL*)webpageURL {
   currentActivity_ = base::scoped_nsobject<NSUserActivity>(
       [[NSUserActivity alloc] initWithActivityType:type]);
   [currentActivity_ setUserInfo:userInfo];
+
+  // Only set the webpageURL if it is valid
+  if (webpageURL && [webpageURL scheme] && [webpageURL host])
```

Comment:
We probably don't need the check anymore, `webpageURL` is guaranteed to be either `nil` or a valid URL, and both are safe to be passed to `setWebpageURL`.